unified-runtime/source/adapters/cuda/context.hpp

```diff
@@ -190,6 +170,59 @@ struct ur_context_handle_t_ {
   std::mutex Mutex;
   std::vector<deleter_data> ExtendedDeleters;
   std::set<ur_usm_pool_handle_t> PoolHandles;
+
+  // Create UMF CUDA memory pool for the host memory (UMF_MEMORY_TYPE_HOST)
+  ur_result_t createHostMemoryPool() {
+    umf_memory_provider_handle_t memoryProviderHost = nullptr;
+    ur_result_t URResult =
+        umf::createHostMemoryProvider(Devices[0], &memoryProviderHost);
+    if (URResult != UR_RESULT_SUCCESS) {
+      return URResult;
+    }
+
+    umf_result_t UmfResult =
+        umfPoolCreate(umfProxyPoolOps(), memoryProviderHost, nullptr,
+                      UMF_POOL_CREATE_FLAG_OWN_PROVIDER, &MemoryPoolHost);
+    UMF_RETURN_UR_ERROR(UmfResult);
+
+    return UR_RESULT_SUCCESS;
+  }
+
+  // Create UMF CUDA memory pools for the device memory (UMF_MEMORY_TYPE_HOST)
+  // and the shared memory (UMF_MEMORY_TYPE_SHARED)
+  ur_result_t createDeviceMemoryPools() {
```

Comment:
Is there a reason you can't just store a `ur_usm_pool_handle_t_` in the context? The ctor of `ur_usm_pool_handle_t_` does pretty much exactly the same thing as this and `createHostMemoryPool` function.

Reply:
`ur_usm_pool_handle_t_::ur_usm_pool_handle_t_` stores user **Disjoint Pools** in the **set** of pools `std::set<ur_usm_pool_handle_t> PoolHandles;`. But I want to store the default **Proxy Pools** in the **vectors**
```c
  std::vector<umf_memory_pool_handle_t> MemoryDevicePools;
  std::vector<umf_memory_pool_handle_t> MemorySharedPools;
```

Reply:
Ok, so to me it looks like the ur_usm_pool_handle_t_  is implemented incorrectly. Look at how, urUSM*Alloc is implemented: https://github.com/intel/llvm/blob/sycl/unified-runtime/source/adapters/cuda/usm.cpp#L71 We are using the DevicePool, without even looking at the hDevice param. The `ur_usm_pool_handle_t_  ` should consist of a vector of memory pools for device and shared allocations just as you implemented in context right now.

I know fixing this, will require a bit more work but it would make the code much simpler if we could just store `ur_usm_pool_handle_t_ ` in the context (not to mention that right now, custom memory pools doesn't seem to work correctly). I won't block this PR but this should be fixed sooner rather than later.